private/check/check.go

```diff
@@ -28,6 +32,27 @@ func NewTestFixtureGetter(fsys fs.FS) *TestFixtureGetter {
 	}
 }
 
+func (g *TestFixtureGetter) PreCacheTestFixtures() error {
+	err := fs.WalkDir(g.fsys, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !d.IsDir() {
+			return nil
+		}
+
+		// We need to search the filesystem for any directory `**/testdata`, omitting nested matches, e.g. `**/testdata/**/testdata`
+		if d.Name() == util.TestDataDirectory && !strings.Contains(filepath.Dir(path), fmt.Sprintf("/%s/", util.TestDataDirectory)) {
+			g.LoadTestFixture(path)
+		}
```

Comment:
Rather than checking for ancestor `testdata`s, we could just not enter the directory:

```suggestion
		if d.Name() == util.TestDataDirectory {
			g.LoadTestFixture(path)
			return fs.SkipDir
		}
```

Reply:
Much nicer, thanks